mobileapps/views.py

```diff
@@ -612,3 +622,238 @@ def _create_notification_message(app_id, payload):
+            except ImageValidationError as error:
+                return Response(
+                    {"developer_message": error.message, "user_message": error.user_message},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            logo_image_names = get_logo_image_names(organization.name)
+            create_logo_images(uploaded_file, logo_image_names)
+
+            # update the organization to reflect that a logo image is available against this theme.
+            set_has_logo_image(organization.id, True, _make_upload_dt())
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class OrganizationThemeDetailView(MobileRetrieveUpdateDestroyAPIView):
+    """
+    **Use Case**
+
+        Get theme for organization, update or destroy the existing theme.
+
+    **Example Requests**
+
+        GET /api/server/mobileapps/organization/{id}/themes/{id}
+        PUT /api/server/mobileapps/organization/{id}/themes/{id}
+        DELETE /api/server/mobileapps/organization/{id}/themes/{id}
+
+        **PUT Parameters**
+
+        The body of the POST request must include the following parameters.
+
+        * file: Image file to be updated as a logo of the theme
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+
+    **Response Values**
+
+        **GET**
+
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+
+        The HTTP 200 response has an object with the following values.
+
+        * id: ID of the mobile app.
+        * created: Datetime it was created in.
+        * modified: Datetime it was modified in.
+        * logo_image:
+            - has_image
+            - image_url_full
+            - image_url_large
+            - image_url_small
+            - image_url_xsmall
+            - image_url_medium
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+        * organization_id: organization id to which theme is related
+
+        **PUT**
+
+        If the request is successful, the request returns an HTTP 204 response.
+
+        * id: ID of the mobile app.
+        * created: Datetime it was created in.
+        * modified: Datetime it was modified in.
+        * logo_image:
+            - has_image
+            - image_url_full
+            - image_url_large
+            - image_url_small
+            - image_url_xsmall
+            - image_url_medium
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+        * organization_id: organization id to which theme is related
+    """
+
+    serializer_class = ThemeSerializer
+    queryset = Theme.objects.all()
+
+
+    def put(self, request, pk, organization_id):
+        data = request.data.copy()
+        data["organization_id"] = organization_id
+
+        theme = Theme.objects.get(pk=pk)
+
+        theme_serializer = ThemeSerializer(theme, data=data)
```

Comment:
@ziafazal Below this line, code is duplicate of the POST view. I am thinking of moving this as a static method against model, a utility method in a separate utils file or make a separate API call of uploading/updating the logo image only in both (POST and PUT) cases. Need your suggestions on these above mentioned options.

Reply:
@aamishbaloch we can create a method i.e `_save_logo` in views.py which should handle the logic around saving log image and then use that method in both POST and PUT handlers.